### ISSUE TITLE
Simplify componentDidUpdate

### ISSUE THREAD

**Issue by albertogasparin** [orgsync/react-list#166]
I was trying to figure out why ReactList starts rendering the list using the `pageSize` setting, but then it re-renders the list again `n` amount of times to "stabilise", increasing the `size` until it eventually implodes (this is what happens with elements animating from `height:0`, see it here: https://jsfiddle.net/4cqqfmqd/)

This is the current implementation:
```js
componentDidUpdate() {
    // If the list has reached an unstable state, prevent an infinite loop.
    if (this.unstable) return;

    if (++this.updateCounter > MAX_SYNC_UPDATES) {
      this.unstable = true;
      return console.error(UNSTABLE_MESSAGE);
    }

    if (!this.updateCounterTimeoutId) {
      this.updateCounterTimeoutId = setTimeout(() => {
        this.updateCounter = 0;
        delete this.updateCounterTimeoutId;
      }, 0);
    }

    this.updateFrame();
}
```

Is there any particular reason why ReactList should run `updateFrame` even if it has just been called?
Why not checking if `updateFrame` was just called (by checking if `state` has just been changed) and if it is so, do nothing?

```js
componentDidUpdate(prevProps, prevState) {
    if (prevState === this.state) {
        this.updateFrame();
    }
}
```
In my tests this prevents the infinite loop and also solves the issue with the above jsfiddle. What do you think?

**Comment by albertogasparin** [comment]
I've done a bit more testing and calling `updateFrame()` only on props change is not enough. So, what do you think about an optional `stableFrameDelay` prop? The implementation could look like this:

```js
componentDidUpdate() {
    // code to prevent infinite loop...
    
    if (this.props.stableFrameDelay) {
        this.updateFrameTimer = setTimeout(() => this.updateFrame(), this.props.stableFrameDelay);
    } else {
        this.updateFrame();
    }
}
```

And on `componentWillReceiveProps` we clear the timer to avoid unnecessary updates.

**Comment by caseywebdev** [comment]
Yes I think something like your `stableFrameDelay` would be the appropriate approach for supporting same-axis size animation, where that value is your animation duration or longer.

**Comment by albertogasparin** [comment]
Cool, if you like this approach would you like me to make a PR?

**Comment by caseywebdev** [comment]
Sure, we can discuss implementation there.
